lib/mongoid/contextual/mongo.rb

```diff
@@ -71,7 +71,31 @@ def cached?
       # @since 3.0.0
       def count(options = {}, &block)
         return super(&block) if block_given?
-        try_cache(:count) { view.count(options) }
+        try_cache(:count) { view.count_documents(options) }
+      end
+
+      # Get the estimated number of documents matching the query.
+      #
+      # @example Get the estimated number of matching documents.
+      #   context.estimated_count
+      #
+      # @example Get the count of documents with the provided options.
+      #   context.estimated_count(limit: 1)
+      #
+      # @example Get the count for where the provided block is true.
+      #   context.estimated_count do |doc|
+      #     doc.likes > 1
+      #   end
```

Comment:
This usage with a block is also not tested as far as I can tell.

Reply:
Emily and I decided not to allow for the block notation with `estimated_count`. The reason `count` has a block is because it is a function often used on `Enumerable` and the user expects to be able to pass in a block, the however is not true with `estimated_count`.

If the user wants to use a filter for an `estimated_count` they can call the `estimated_count` method on a criteria that uses that filter.